common/persistence/execution_manager_test.go

```diff
@@ -480,6 +480,159 @@ func TestExecutionManager_UpdateWorkflowExecution(t *testing.T) {
 	assert.Equal(t, stats, res.MutableStateUpdateSessionStats)
 }
 
+func TestSerializeWorkflowSnapshot(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		prepareMocks func(*MockPayloadSerializer)
+		input        *WorkflowSnapshot
+		checkRes     func(*testing.T, *InternalWorkflowSnapshot, error)
+	}{
+		{
+			name: "success",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeEvent(gomock.Any(), gomock.Any()).Return(sampleEventData(), nil).Times(5)
```

Comment:
Maybe we should have a comment on why we expect this 5 times.

Reply:
Switched to explicit 5 assertions so it would be easier to debug if something is missing.